waggle-dance-core/src/test/java/com/hotels/bdp/waggledance/mapping/service/impl/PrefixBasedDatabaseMappingServiceTest.java

```diff
@@ -275,6 +271,65 @@ public void close() throws IOException {
+    when(metaStoreMappingFederated.transformInboundDatabaseName(DB_PREFIX + FEDERATED_DB)).thenReturn(FEDERATED_DB);
+    federatedMetastore.setMappedDatabases(Collections.singletonList(FEDERATED_DB));
+    MappedTables mappedTables = new MappedTables(FEDERATED_DB, Lists.newArrayList("table"));
+    federatedMetastore.setMappedTables(Collections.singletonList(mappedTables));
+    service = new PrefixBasedDatabaseMappingService(metaStoreMappingFactory,
+        Arrays.asList(primaryMetastore, federatedMetastore), queryMapping);
+    DatabaseMapping mapping = service.databaseMapping(DB_PREFIX + FEDERATED_DB);
+    service.checkTable(DB_PREFIX + FEDERATED_DB, "table_not_mapped", mapping);
+  }
+
+  @Test(expected = NoSuchObjectException.class)
+  public void checkTableMappedTablesConfigPresentPrimary() throws NoSuchObjectException {
+    primaryMetastore.setMappedDatabases(Collections.singletonList(PRIMARY_DB));
+    MappedTables mappedTables = new MappedTables(PRIMARY_DB, Lists.newArrayList("table"));
+    primaryMetastore.setMappedTables(Collections.singletonList(mappedTables));
+    service = new PrefixBasedDatabaseMappingService(metaStoreMappingFactory,
+        Arrays.asList(primaryMetastore, federatedMetastore), queryMapping);
+    DatabaseMapping mapping = service.databaseMapping(PRIMARY_DB);
+    service.checkTable(PRIMARY_DB, "table_not_mapped", mapping);
+  }
+
+  @Test
+  public void checkTableMappedTablesAllowed() throws NoSuchObjectException {
+    String otherDb = "other_db";
+    primaryMetastore.setMappedDatabases(Lists.newArrayList(PRIMARY_DB, otherDb));
+    MappedTables mappedTables1 = new MappedTables(PRIMARY_DB, Lists.newArrayList("table"));
+    MappedTables mappedTables2 = new MappedTables(otherDb, Lists.newArrayList("table1"));
+    primaryMetastore.setMappedTables(Lists.newArrayList(mappedTables1, mappedTables2));
+    service = new PrefixBasedDatabaseMappingService(metaStoreMappingFactory,
+        Arrays.asList(primaryMetastore, federatedMetastore), queryMapping);
+    DatabaseMapping mapping = service.databaseMapping(PRIMARY_DB);
+    service.checkTable(PRIMARY_DB, "table", mapping);
+    mapping = service.databaseMapping(otherDb);
+    service.checkTable(otherDb, "table1", mapping);
+  }
+
+  @Test
+  public void filterTables() throws NoSuchObjectException {
+    List<String> allowedTables = Lists.newArrayList("table", "another_table");
+    primaryMetastore.setMappedDatabases(Collections.singletonList(PRIMARY_DB));
+    MappedTables mappedTables = new MappedTables(PRIMARY_DB, allowedTables);
+    primaryMetastore.setMappedTables(Collections.singletonList(mappedTables));
+    service = new PrefixBasedDatabaseMappingService(metaStoreMappingFactory,
+        Arrays.asList(primaryMetastore, federatedMetastore), queryMapping);
+    DatabaseMapping mapping = service.databaseMapping(PRIMARY_DB);
+    List<String> result = service.filterTables(PRIMARY_DB,
```

Comment:
Ah, OK, this now makes it clear to me what `filterTables` does, it filters out those that don't pass the allow list. I think just adding javadoc on the interface to make it a bit clearer is all that's needed here to address my concerns.